docs/aws/audit/redshiftmonitoring/rules/emr_kerberos_enabled.mdx

```diff
@@ -23,6 +23,260 @@ CBP,RBI_MD_ITF
+         "Resource": "*",
+         "Condition": {
+           "StringNotEquals": {
+             "elasticmapreduce:SecurityConfiguration": "MyKerberosConfig"
+           }
+         }
+       }
+     ]
+   }'
+   ```
+
+4. **Attach the IAM Policy to Users or Roles:**
+   Attach the policy to the IAM users or roles that are responsible for creating EMR clusters to ensure compliance.
+
+   ```sh
+   aws iam attach-user-policy --user-name MyUser --policy-arn arn:aws:iam::123456789012:policy/EnforceKerberosPolicy
+   ```
+
+By following these steps, you can ensure that all Amazon EMR clusters are created with Kerberos enabled, thereby preventing misconfigurations related to security.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Amazon EMR clusters from being created without Kerberos enabled using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+### Step 1: Install Boto3
+First, ensure you have Boto3 installed. You can install it using pip if you haven't already:
+```bash
+pip install boto3
+```
+
+### Step 2: Set Up AWS Credentials
+Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by creating a `~/.aws/credentials` file.
+
+### Step 3: Create a Python Script to Check and Enable Kerberos
+Here's a Python script that checks if Kerberos is enabled for an EMR cluster and enables it if not:
+
+```python
+import boto3
+from botocore.exceptions import ClientError
+
+# Initialize a session using Amazon EMR
+emr_client = boto3.client('emr')
+
+def enable_kerberos(cluster_id):
+    try:
+        # Describe the cluster to get its current configuration
+        response = emr_client.describe_cluster(ClusterId=cluster_id)
+        security_configuration = response['Cluster']['SecurityConfiguration']
+
+        if 'KerberosAttributes' not in security_configuration:
+            print(f"Kerberos is not enabled for cluster {cluster_id}. Enabling Kerberos...")
+
+            # Define Kerberos attributes
+            kerberos_attributes = {
+                'Realm': 'YOUR.REALM',
+                'KdcAdminPassword': 'YOUR_KDC_ADMIN_PASSWORD',
+                'CrossRealmTrustPrincipalPassword': 'YOUR_CROSS_REALM_TRUST_PRINCIPAL_PASSWORD',
+                'ADDomainJoinUser': 'YOUR_AD_DOMAIN_JOIN_USER',
+                'ADDomainJoinPassword': 'YOUR_AD_DOMAIN_JOIN_PASSWORD'
+            }
+
+            # Enable Kerberos
+            emr_client.modify_cluster(
+                ClusterId=cluster_id,
+                KerberosAttributes=kerberos_attributes
+            )
+            print(f"Kerberos has been enabled for cluster {cluster_id}.")
+        else:
+            print(f"Kerberos is already enabled for cluster {cluster_id}.")
+
+    except ClientError as e:
+        print(f"An error occurred: {e}")
+
+# Example usage
+cluster_id = 'j-XXXXXXXXXXXXX'  # Replace with your cluster ID
+enable_kerberos(cluster_id)
+```
+
+### Step 4: Automate the Script Execution
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>